src/windows/webview.c

```diff
@@ -477,21 +478,23 @@ static ICoreWebView2EnvironmentOptionsVtbl VTBL5 = {
 
 // Public
 
-static HMODULE webview_load_dll(void)
+static bool webview_dll_path(WCHAR *pathw, bool as_user)
 {
+	bool ok = false;
+
 	HKEY key = NULL;
-	WCHAR *dll = NULL;
-	HMODULE lib = NULL;
+	WCHAR dll[MTY_PATH_MAX] = {0};
+
+	// https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/distribution#detect-if-a-webview2-runtime-is-already-installed
+	const wchar_t *machine_path = L"Software\\Microsoft\\EdgeUpdate\\ClientState\\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}";
+	const wchar_t *user_path = L"Software\\Microsoft\\EdgeUpdate\\Clients\\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}";
+	const wchar_t *reg_path = as_user ? user_path : machine_path;
 
-	LSTATUS r = RegOpenKeyEx(HKEY_LOCAL_MACHINE,
-		L"Software\\Microsoft\\EdgeUpdate\\ClientState\\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}",
-		0, KEY_WOW64_32KEY | KEY_READ, &key);
+	LSTATUS r = RegOpenKeyEx(HKEY_LOCAL_MACHINE, reg_path, 0, KEY_WOW64_32KEY | KEY_READ, &key);
```

Comment:
The page you linked states that the per-user webview2 runtime install location belongs to the `HKEY_CURRENT_USER` registry hive, whereas your code concatenates the `user_path` to the `HKEY_LOCAL_MACHINE` hive. 

Am I missing something?

![image](https://github.com/user-attachments/assets/0a9eb513-aaea-4109-ba58-963f21ee08b6)